internal/reader/generate/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"log/slog"
+	"net/http"
+	"os"
+
+	"github.com/calindra/nonodo/internal/commons"
+)
+
+const (
+	urlReader = "https://github.com/cartesi/rollups-node/releases/download/v1.5.0-rc1/schema.graphql"
+)
+
+// Exit if there is any error.
+func checkErr(context string, err any) {
+	if err != nil {
+		log.Fatal(context, ": ", err)
+	}
+}
+
+func downloadSchema(url string) ([]byte, error) {
```

Comment:
What do you think about transferring the responsibility of stopping the application from the downloadSchema function to the main function? Instead of using checkErr directly in downloadSchema, it would only log and return errors. The main function would then decide whether to stop the application based on the returned errors. This centralizes the control flow in main and keeps downloadSchema more focused on its core tasks.